### PR TITLE
[MNT] Updating the GHA dependencies to install OSX dependencies and setting the compiler flags

### PR DESCRIPTION
**Pull request by fkiraly**
Combination of https://github.com/sktime/sktime/pull/6926 and https://github.com/sktime/sktime/pull/6933

Also same as revert of sktime/sktime#6935.

### CODE REVIEW

**Comment by yarnabrina** [comment]
Just for my curiosity, can I ask what is happening with the `libomp` related changes @fnhirwa made in Darts PR? It was tested there and if I remember correctly, all jobs passed (subject to differential testing). Then it is merged separately and reverted and this PR seems to make again, and I am very confused.

**Comment by fkiraly** [comment]
I asked the `script` folder to be moved to within `.github` to not clutter the root, which @fnhirwa did. Unfortunately he forgot to change the references, which he fixed in #6933, but that still was failing.

To avoid major breakage on main, I reverted, and hope that I (or @fnhirwa) can figure out why #6933 did not fix it, it's probably a typo somewhere because in the `darts` PR it originally worked, as you say.

If you want to have a look, that would also be appreciated! It's probably sth silly and small.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -197,8 +197,9 @@ jobs:
       - name: Display Python version
         run: python -c "import sys; print(sys.version)"
 
-      - name: Install OS packages
-        run: brew install libomp
+      - name: Install OSX packages
+        shell: bash
+        run: ./.github/scripts/install_osx_dependencies.sh
```
Comment (yarnabrina) [inline .github/workflows/test.yml:202]: Line 202 uses `./.github/scripts`.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -284,9 +285,12 @@ jobs:
       - name: Display Python version
         run: python -c "import sys; print(sys.version)"
 
+      - name: Install OSX packages
+        run: ./scripts/install_osx_dependencies.sh
```
Comment (yarnabrina) [inline .github/workflows/test.yml:289]: Line 289 uses `./scripts`.

**Review by yarnabrina — COMMENTED** [review]
Not sure if this is the reason for failures, but I notice this difference between two line of `.github/workflows/test.yml`. It may or may not matter.

File: .github/workflows/test.yml (hunk 2 above)
Comment (fkiraly) [reply .github/workflows/test.yml:289]: that's probably it! The correct one is for the workflow that does not break in the diagnostic PR, I reckon...

**Comment by fkiraly** [comment]
looks like this fixed it, it was just a typo worth a single `.`, @fnhirwa!

File: .github/workflows/test.yml (hunk 2 above)
Comment (fnhirwa) [reply .github/workflows/test.yml]: nice catch. I couldn't figure this one out 🤣

**Pull request merged**